flow/project.py

```diff
@@ -1087,7 +1089,86 @@ def print_status(self, jobs=None, overview=True, overview_max_lines=None,
         if jobs is None:
             jobs = self     # all jobs
 
-        tmp = self._fetch_status(jobs, err, ignore_errors, no_parallelize)
+        if profile:
+            try:
+                import pprofile
+            except ImportError:
+                raise RuntimeWarning(
+                    "Profiling requires the pprofile package. "
+                    "Install with `pip install pprofile`.")
+            prof = pprofile.StatisticalProfile()
+
+            fn_filter = [
+                inspect.getfile(threading),
+                inspect.getfile(multiprocessing),
+                inspect.getfile(Pool),
+                inspect.getfile(ThreadPool),
+            ]
+
+            with prof(single=False):
+                tmp = self._fetch_status(jobs, err, ignore_errors, no_parallelize)
+
+            prof._mergeFileTiming()
+
+            # Unrestricted
+            total_impact = 0
+            total_num_hits = sum([hit[2] for fn, ft in prof.merged_file_dict.items()
+                                  if fn not in fn_filter for hit in ft.iterHits()])
+            hits = [hit for fn, ft in prof.merged_file_dict.items()
+                    if fn not in fn_filter for hit in ft.iterHits()]
+            sorted_hits = reversed(sorted(hits, key=lambda hit: hit[2]))
+
+            profiling_results = [
+                '# Profiling:\n',
+                'Rank Impact Code object',
+                '---- ------ -----------']
```

Comment:
Could/should we jinja-ify this?

Reply:
Yes, I'll rebase this once #111 is merged.